xcdat/regridder/regrid2.py

```diff
@@ -486,21 +402,46 @@ def _pertub(value: xr.DataArray) -> xr.DataArray:
 
     Parameters
     ----------
-    value : xr.DataArray
+    value : np.ndarray
         Value to pertub.
 
     Returns
     -------
-    xr.DataArray
+    np.ndarray
         Value that's been pertubed.
     """
     if value >= 0.0:
         offset = np.ceil(value + 0.000001)
     else:
         offset = np.floor(value - 0.000001) + 1.0
 
-    return xr.DataArray(offset)
+    return offset
 
 
 # vectorize version of pertub
 _vpertub = np.vectorize(_pertub)
+
+
+def _get_dimension(input_data_var, cf_axis_name):
+    name = input_data_var.cf.axes[cf_axis_name]
+
+    if isinstance(name, list):
+        name = name[0]
+
+    index = input_data_var.dims.index(name)
+
+    return name, index
+
+
+def _get_bounds_ensure_dtype(ds, axis):
+    try:
+        name = ds.cf.bounds[axis][0]
+    except (KeyError, IndexError):
+        raise RuntimeError(f"Could not determine {axis!r} bounds")
```

Comment:
```suggestion
    try:
        name = ds.bounds.get_bounds(axis)
    except (ValueError, KeyError):
        raise RuntimeError(f"Could not determine {axis!r} bounds")
```

I think you can use xCDAT's `ds.bounds.get_bounds()`.

Reply:
As from above I could use `get_bounds` but we're again trading performance for robustness.

Reply:
Gotcha. If the current implementation is faster and using `.get_bounds()` isn't necessary, we can keep your current implementation.

Reply:
After some review I'm going to leave the `ds.cf.bounds` usage. In this case it's fine to use as it's being called on the input/output grid, both of which have been generated/validated by previous code. We can guarantee that both grid objects only have lat/lon coordinates/bounds with the correct metadata thus we do not need the robustness of `ds.bounds.get_bounds`.